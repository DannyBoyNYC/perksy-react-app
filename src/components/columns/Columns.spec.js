const React = require('react');

const Columns = require('./index').default;

describe('<Columns.* />', () => {
  let component;
  const requiredProps = {
    children: <div className="content" />
  };

  describe('<Columns.Even />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Columns.Even {...requiredProps} />,
        'Columns'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('adds a `wrap` class if shouldWrap is true', () => {
      component.setProps({
        shouldWrap: true
      });
      expect(component).toHaveClassName('wrap');
    });
  });

  describe('<Columns.OneSpanTwo />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Columns.OneSpanTwo {...requiredProps} />,
        'Columns'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Columns.SpanTwoOne />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Columns.SpanTwoOne {...requiredProps} />,
        'Columns'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Columns.SpanThreeSpanTwo />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Columns.SpanThreeSpanTwo {...requiredProps} />,
        'Columns'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
