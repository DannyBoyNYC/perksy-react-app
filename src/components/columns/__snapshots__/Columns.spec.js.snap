// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Columns.* /> <Columns.Even /> should render correctly with all required props 1`] = `
<div
  className="Columns-root-0-1-1 Columns-even-0-1-2"
>
  <div
    className="column"
    key=".0"
  >
    <div
      className="content"
    />
  </div>
</div>
`;

exports[`<Columns.* /> <Columns.OneSpanTwo /> should render correctly with all required props 1`] = `
<div
  className="Columns-root-0-1-1 Columns-one-span-two-0-1-3"
>
  <div
    className="column"
    key=".0"
  >
    <div
      className="content"
    />
  </div>
</div>
`;

exports[`<Columns.* /> <Columns.SpanThreeSpanTwo /> should render correctly with all required props 1`] = `
<div
  className="Columns-root-0-1-1 Columns-span-three-span-two-0-1-5"
>
  <div
    className="column"
    key=".0"
  >
    <div
      className="content"
    />
  </div>
</div>
`;

exports[`<Columns.* /> <Columns.SpanTwoOne /> should render correctly with all required props 1`] = `
<div
  className="Columns-root-0-1-1 Columns-span-two-one-0-1-4"
>
  <div
    className="column"
    key=".0"
  >
    <div
      className="content"
    />
  </div>
</div>
`;
